pyart/retrieve/qpe.py

```diff
@@ -1,18 +1,695 @@
+    else:
+        is_slave = rain_master['data'] < thresh
+
+    rain_master['data'][is_slave] = rain_slave['data'][is_slave]
+
+    rr_data[is_lr] = rain_master['data'][is_lr]
+    rr_data[is_rn] = rain_master['data'][is_rn]
+
+    # mixed phase
+    rr_data[is_ws] = mp_factor*rain_z['data'][is_ws]
+    rr_data[is_mh] = mp_factor*rain_z['data'][is_mh]
+
+    rain = get_metadata(rr_field)
+    rain['data'] = rr_data
+
+    return rain
+
+
+def _get_coeff_rkdp(freq):
+    """
+    get the R(kdp) power law coefficients for a particular frequency
+
+    Parameters
+    ----------
+    freq : float
+        radar frequency [Hz]
+
+    Returns
+    -------
+    alpha, beta : floats
+        the coefficient and exponent of the power law
+
+    """
+    coeff_rkdp_dict = _coeff_rkdp_table()
+
+    freq_band = get_freq_band(freq)
+    if (freq_band is not None) and (freq_band in coeff_rkdp_dict):
+        return coeff_rkdp_dict[freq_band]
+
+    if freq < 2e9:
+        freq_band_aux = 'S'
+    elif freq > 12e9:
+        freq_band_aux = 'X'
+
+    warn('Radar frequency out of range. ' +
+         'Coefficients only applied to S, C or X band. ' +
+         freq_band + ' band coefficients will be used')
+
+    return coeff_rkdp_dict[freq_band_aux]
+
+
+def _coeff_rkdp_table():
+    """
+    defines the R(kdp) power law coefficients for each frequency band.
+
+    Returns
+    -------
+    coeff_rkdp_dict : dict
+        A dictionary with the coefficients at each band
+
+    """
+    coeff_rkdp_dict = dict()
+
+    # S band: Beard and Chuang coefficients
+    coeff_rkdp_dict.update({'S': (50.70, 0.8500)})
+
+    # C band: Beard and Chuang coefficients
+    coeff_rkdp_dict.update({'C': (29.70, 0.8500)})
+
+    # X band: Brandes coefficients
+    coeff_rkdp_dict.update({'X': (15.81, 0.7992)})
+
+    return coeff_rkdp_dict
+
+
+def _get_coeff_ra(freq):
+    """
+    get the R(A) power law coefficients for a particular frequency
+
+    Parameters
+    ----------
+    freq : float
+        radar frequency [Hz]
+
+    Returns
+    -------
+    alpha, beta : floats
+        the coefficient and exponent of the power law
+
+    """
+    coeff_ra_dict = _coeff_ra_table()
+
+    freq_band = get_freq_band(freq)
+    if (freq_band is not None) and (freq_band in coeff_ra_dict):
+        return coeff_ra_dict[freq_band]
+
+    if freq < 2e9:
+        freq_band_aux = 'S'
+    elif freq > 12e9:
+        freq_band_aux = 'X'
+
+    warn('Radar frequency out of range. ' +
+         'Coefficients only applied to S, C or X band. ' +
+         freq_band + ' band coefficients will be used')
+
+    return coeff_ra_dict[freq_band_aux]
+
+
+def _coeff_ra_table():
+    """
+    defines the R(A) power law coefficients for each frequency band.
+
+    Returns
+    -------
+    coeff_ra_dict : dict
+        A dictionary with the coefficients at each band
+
+    """
+    coeff_ra_dict = dict()
+
+    # S band: at 10�C according to tables from Ryzhkov et al. 2014
```

Comment:
Spell out degrees.  I would prefer not to introduce unicode into the source files unless necessary